File: Domain/Entities/Curriculum/SyllabusElements/Lesson.cs
using Domain.Entities.Curriculum.LearningElements.Interfaces;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Domain.Entities.Curriculum.SyllabusElements;

/// <summary>
/// A tree of learning elements
/// </summary>
public class Lesson
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;

    public Lesson? Prev { get; set; }
    public HashSet<Lesson> Next { get; set; } = new();


    public ILearningElement StartingElement { get; set; }

    public Lesson(string name)
    {
        Name = name;
    }
}

File: Domain/Entities/Curriculum/SyllabusElements/Lesson.java

import java.util.HashSet;

public class Lesson {
    private int id;
    private String name = "";
    private Lesson prev;
    private HashSet<Lesson> next = new HashSet<Lesson>();
    private ILearningElement startingElement;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Lesson getPrev() {
        return prev;
    }

    public void setPrev(Lesson prev) {
        this.prev = prev;
    }

    public HashSet<Lesson> getNext() {
        return next;
    }

    public void setNext(HashSet<Lesson> next) {
        this.next = next;
    }

    public ILearningElement getStartingElement() {
        return startingElement;
    }

    public void setStartingElement(ILearningElement startingElement) {
        this.startingElement = startingElement;
    }

    public Lesson(String name) {
        this.name = name;
    }

    public getLerningElem(LerningElem elem){
        HashSet<LerningElem> ElemToDo = elem.getNext();
    }


}
